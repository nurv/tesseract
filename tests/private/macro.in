2 + 3
macro /([0-9]+)\s*\+/ { $1$ * }
2 + 3
ist.leic.pa.utils.MacroManager.macros();
ist.leic.pa.utils.MacroManager.deleteMacro(1);
2 + 3
macro /entity ([^;]*);/ { block{
int $1$ = 0;
String &{ "get" + "$1$".substring(0, 1).toUpperCase() + 
	   "$1$".substring(1).toLowerCase() } (){
	return %{ "$1$" + " is :"} + $1$;
}
}}
entity knight;
knight
getKnight()
