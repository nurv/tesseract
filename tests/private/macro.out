> 5
> > 6
> 1 : macro ([0-9]+)\s*\+ => $1$ *
> > 5
> > > 0
> "knight is :0"
> 